src/features/query-bar/components/filter/Condition.js

```diff
@@ -201,48 +179,59 @@ const Condition = ({
     };
 
     const renderOperatorField = () => {
-        const { operatorKey } = condition;
+        const { operator } = condition;
         const columnOperators = getColumnOperators();
-        const operatorOptions = getFormattedOptions(columnOperators);
+        const operatorOptions = columnOperators.map(_operator => {
+            const { displayText, key } = _operator;
+            return {
+                displayText,
+                value: key,
+            };
+        });
 
         return (
             <div className="condition-operator-dropdown-container">
                 <div className="filter-dropdown-single-select-field-container">
                     <SingleSelectField
                         fieldType={OPERATOR}
                         isDisabled={false}
-                        onChange={updateSelectedField}
+                        onChange={updateOperatorField}
                         options={operatorOptions}
-                        selectedValue={operatorKey}
+                        selectedValue={operator}
                     />
                 </div>
             </div>
         );
     };
 
     const renderValueField = () => {
-        const { valueKey, valueType } = condition;
+        const { columnId, value } = condition;
 
-        const columnOptions = getColumnOptions();
-        const valueOptions = getFormattedOptions(columnOptions);
-        const error = getErrorMessage();
+        const column = columns && columns.find(c => c.id === columnId);
+        const type = column && column.type;
 
-        const classnames = classNames('condition-value-dropdown-container', {
-            'show-error': error,
-        });
+        if (column && type) {
+            const valueOptions = getColumnOptions();
+            const error = getErrorMessage();
 
-        return (
-            <div className={classnames}>
-                <ValueField
-                    selectedValue={valueKey}
-                    updateValueField={updateValueField}
-                    updateSelectedField={updateSelectedField}
-                    valueKey={valueKey}
-                    valueOptions={valueOptions}
-                    valueType={valueType}
-                />
-            </div>
-        );
+            const classnames = classNames('condition-value-dropdown-container', {
+                'show-error': error,
+            });
+
+            return (
+                <div className={classnames}>
+                    <ValueField
+                        selectedValue={value}
+                        updateValueField={updateValueField}
+                        updateSelectedField={updateSelectedField}
```

Comment:
The reason we have updateValueField and updateSelectedField (even though both contain very similar logic) is that TextInput passes up a param which is a `SyntheticEvent` but `SingleSelectField` passes up an `OptionType` (which is something that we defined).